components/tidb_query/src/rpn_expr/impl_compare.rs

```diff
@@ -227,6 +227,29 @@ pub fn coalesce<T: Evaluable>(args: &[&Option<T>]) -> Result<Option<T>> {
     Ok(None)
 }
 
+#[rpn_fn(varg)]
+#[inline]
+pub fn interval_real(args: &[&Option<Real>]) -> Result<Option<Int>> {
+    let target = match args[0] {
+        None => return Ok(Some(-1)),
+        Some(v) => *v,
+    };
+
+    let mut left = 1;
```

Comment:
Would you mind using [slice::binary_search_by](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.binary_search_by) or [slice::binary_search_by_key](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.binary_search_by_key)?